### PR TITLE
fix (faviconfeed): domains instead of url/urls in remote tippy top manifest

### LINKED ISSUE #3855: Use domains instead of urls from tippy endpoint
Somewhat depends on #3847 but this can land before and just happen to do nothing until the endpoint is updated.

### PR DESCRIPTION
**Pull request by rlr**
Fixes #3855 

@Mardak r?

Can set the pref to https://raw.githubusercontent.com/rlr/tippy-top-playground/master/data/tippy-top-domains.json for testing with airbnb.com and go.twitch.tv.

### CODE REVIEW

File: system-addon/lib/FaviconFeed.jsm [hunk 1]
```diff
@@ -92,8 +92,8 @@ this.FaviconFeed = class FaviconFeed {
     for (const site of sites) {
       // The tippy top manifest can have a url property (string) or a
       // urls property (array of strings)
-      for (const url of site.url ? [site.url] : site.urls || []) {
-        sitesByDomain[getDomain(url)] = site;
+      for (const domain of site.domains || []) {
+        sitesByDomain[domain] = site;
```
Comment (Mardak) [inline system-addon/lib/FaviconFeed.jsm]: getDomain was stripping off the `www.` too, so I guess we just need to make sure we do that for the manifest

Comment (Mardak) [reply system-addon/lib/FaviconFeed.jsm]: Noted on irc:
maybe be explicit when setting `sitesByDomain[domain] = {image_url: site.image_url};` ?

File: system-addon/lib/ActivityStream.jsm [hunk 2]
```diff
@@ -137,7 +137,7 @@ const PREFS_CONFIG = new Map([
   }],
   ["tippyTop.service.endpoint", {
     title: "Tippy Top service manifest url",
-    value: "https://activity-stream-icons.services.mozilla.com/v1/icons.json.br"
+    value: "https://raw.githubusercontent.com/rlr/tippy-top-playground/master/data/tippy-top-domains.json"
```
Comment (Mardak) [inline system-addon/lib/ActivityStream.jsm]: This should be unchanged

**Comment by rlr** [comment]
OK. last sanity check ^

**Comment by as-pine-proxy** [comment]
[Treeherder: 67b1ef1a78186bb3cb1949f8e8515e383dd549db](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=67b1ef1a78186bb3cb1949f8e8515e383dd549db)

**Pull request merged**